File: app/src/main/java/at/devfest/app/core/dagger/module/AppModule.java
package at.devfest.app.core.dagger.module;

import android.app.Application;

import javax.inject.Singleton;

import at.devfest.app.DevFestApp;
import dagger.Module;
import dagger.Provides;

@Module
public final class AppModule {

    private final DevFestApp app;

    public AppModule(DevFestApp app) {
        this.app = app;
    }

    @Provides
    @Singleton
    Application provideApplication() {
        return app;
    }
}

File: app/src/main/java/at/devfest/app/core/dagger/module/AppModule.kt
package at.devfest.app.core.dagger.module

import android.app.Application
import at.devfest.app.DevFestApp
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule(private val app: DevFestApp) {

    @Provides
    @Singleton
    internal fun provideApplication(): Application {
        return app
    }
}
